### PR TITLE
feat - add tab search fields

### PR DESCRIPTION
**Pull request by maisaritense**

### CODE REVIEW

File: projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/dossier-management-detail-container.component.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        </ul>
+
+        <div class="card-body bg-white p-5 position-relative tab-container">
+
+           <div *ngIf="isCase">
+            <valtimo-dossier-management-tab-details></valtimo-dossier-management-tab-details>
+          </div>
+
+          <div *ngIf="!isCase">
+            <p>search fields</p>
```
Comment (mritense) [inline projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/dossier-management-detail-container.component.html:25]: Should be your search field component?

File: projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/dossier-management-detail-container.component.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+@Component({
+  selector: 'valtimo-dossier-management-detail-container',
+  templateUrl: './dossier-management-detail-container.component.html',
+  styleUrls: ['./dossier-management-detail-container.component.css']
+})
+export class DossierManagementDetailContainerComponent {
+
+  public isCase= true;
+  public isSearchFields: boolean;
```
Comment (mritense) [inline projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/dossier-management-detail-container.component.ts:27]: Unused property

File: projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/dossier-management-detail-container.component.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  displayBodyComponent(tab: string): void {
+    tab === 'case' ? this.isCase = true : this.isCase = false
```
Comment (mritense) [inline projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/dossier-management-detail-container.component.ts:30]: this.isCase = tab === 'case'

File: projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/tabs/details/details.component.html [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+<div class="main-content pt-0">
+  <div class="container-fluid">
+    <div *ngIf="documentDefinition" class="btn-group mt-m3px mb-3 float-right">
+      <button (click)="downloadDefinition()" class="btn btn-primary btn-space">
+        <i class="icon mdi mdi-download mr-1"></i>
+        {{ 'Download document definition' | translate }}
+      </button>
+      <a id="downloadAnchorElement" style="display: none"></a>
+      <button
+              (click)="openDossierRemoveModal()"
```
Comment (mritense) [inline projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/tabs/details/details.component.html:26]: Run prettier on all files

File: projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/dossier-management-detail-container.component.html (hunk 1 above)
Comment (maisaritense) [reply projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/dossier-management-detail-container.component.html:25]: added the new component <valtimo-dossier-management-search-fields>

File: projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/dossier-management-detail-container.component.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+export class DossierManagementDetailContainerComponent {
+  public isCase = true;
+  readonly showCaseSearchFields!: boolean;
+
+  constructor(private readonly configService: ConfigService) {
+    this.showCaseSearchFields = configService.config.featureToggles.caseSearchFields;
+  }
+
+  displayBodyComponent(tab: string): void {
+    (this.isCase = tab === 'case')
```
Comment (theo-ritense) [inline projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/dossier-management-detail-container.component.ts:34]: The ternary operator (?..:) is not necessary. The assignment should be enough:
`this.isCase = tab === 'case';`

Maybe for readability add some parentheses like this:
`this.isCase = (tab === 'case');`

File: projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/tabs/dossier-management-detail/dossier-management-detail.component.ts [hunk 6]
```diff
@@ -37,15 +36,11 @@ export class DossierManagementDetailComponent implements OnInit {
   @ViewChild('dossierRemoveModal') dossierRemoveModal: DossierManagementRemoveModalComponent;
   @ViewChild('documentRoles') documentRoles: DossierManagementRolesComponent;
 
-  readonly showCaseSearchFields!: boolean;
-
   constructor(
     private documentService: DocumentService,
     private route: ActivatedRoute,
     private alertService: AlertService,
```
Comment (theo-ritense) [inline projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/tabs/dossier-management-detail/dossier-management-detail.component.ts:42]: please remove the trailing comma

**Review by Rick-Ritense — APPROVED** [review]

**Pull request merged**